### PR TITLE
Roberto-refactored basic information tab and updates timezone array

### PR DESCRIPTION
**Pull request by robertoooc**
# Description
- The timezone.js file contains an array of timezones that is missing a few locations causing the user profile page to crash sometimes due to it not being there when trying the get time zone function.
- The basic information tab also is rendering a tablet view and desktop view simultaneously and not conditionally, this means there are 2 of everything except one version's css hides it while the other is present.

## Related PRS (if any):
Test with latest backend development branch

## Main changes explained:
- updated the timeZones array in the timezones.js file to have over 400 locations now.
- added some error handling for the getOffsetBetweenTimezonesForDate function in the BasicInformationTab.jsx file incase there are any locations missing from the array above, to avoid the page from crashing.
- updated the rendering of the basic information elements to render conditionally based on window size.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to your userprofile → Basic Information Tab→ Location input
6. try entering a few different locations then clicking the get time zone function, notably a few locations that used to crash the page are: vatican, zagreb, podgorica.
7. verify that the locations from step 6 don't crash the page and actually change the timezone accordingly.
8. Take a look at the css properties of the entire basic information tab, compare it to the original versions and make sure it has the same behavior as before since it was refactored. Make sure to change media sizes to verify this.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/632c19c5-75ba-4a65-bbb8-8c4a6a271a55

## Note:
If you notice slow page refresh performance, please note that the performance issues are unrelated to the changes made in this pull request. This pr is tackling a different problem.

## If the get timezone function doesn't work for you : you might be missing the api key
- sign up for an api key at https://opencagedata.com/
- add TIMEZONE_PREMIUM_KEY=your api key to your Backend env variables
- add TIMEZONE_COMMON_KEY= your api key to your Backend env variables
- Make sure to restart backend after changes to .env and try the function again on frontend

### CODE REVIEW

**Review by KevHnh — CHANGES_REQUESTED** [review]
Not sure if this is only happening to me but it appears that nothing happens when I click on "Get Time Zone". The css fix looks great though. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83934825/4a6530a1-1642-4af5-912e-f38b9d962a39

**Comment by robertoooc** [comment]
@KevHnh, 
1. Are you getting any errors in your console? 
2. Check your network requests and look for the get request sent out to the opencage api should look like this `https://api.opencagedata.com/geocode/v1/json?key=apikey&q=vatican&pretty=1&limit=1`, you might be missing the api key which if so: 
- sign up for an api key at https://opencagedata.com 
- add `TIMEZONE_PREMIUM_KEY=your api key` to your Backend env variables
- add `TIMEZONE_COMMON_KEY= your api key` to your Backend env variables
- Make sure to restart backend after changes to .env and try the function again on frontend

**Review by ShengjieMao — APPROVED** [review]
CSS works fine on the page. Also met the problem of having the property unchanged when input different places. But can work after following the adding api steps. The crashed values in step 6 also work fine in tests. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/6681de7e-0905-4d3a-9b90-cf16682cd5ed

File: src/components/UserProfile/BasicInformationTab/BasicInformationTab.css [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
-@media (max-width: 1199px) {
-  .phone-input-style.phone-input-style {
-    width: 265px !important;
-  }
+.phone-input-style.phone-input-style {
+  width: 100%; 
+  max-width: 300px; 
 }
 
 @media (max-width: 1024px) {
```
Comment (Renanluizssx) [inline src/components/UserProfile/BasicInformationTab/BasicInformationTab.css:9]: you could specify that it'll apply to the screens, `@media screen and (max-width: 1024px) {`

**Review by Renanluizssx — CHANGES_REQUESTED** [review]
Hi @robertoooc, i didn't get to add a timezone, because when i click in the "get time zone", nothing happens.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/ab9dc91b-25d8-4631-82e9-1290d00c6d18

**Review by SRamen1999 — CHANGES_REQUESTED** [review]
After reviewing this PR I noticed a few problems.

When changing the window size to be smaller it makes the website turn into a blank white screen. Even if you refresh the page it will still be a white screen. But if you increase the window size then refresh it goes back to normal. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/89ee7f21-8bab-4f96-a403-61b176a330f5

Another Issue I came across is, nothing happens when I click on Get Time Zone. I even added the missing api keys.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/51046888-cf1e-4040-ab89-5b4aa5733df5

Screen Shot 2024-01-25 at 6 35 29 PM

**Review by pshereen — APPROVED** [review]
Tested the PR and all the updates are working as per the requirement.

Changed the get time zone function to Vatican, Zagreb, Podgorica, and is working without crashing the page.

Checked CSS properties of the basic information tab in different media sizes and it is rendering properly based on window size.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/9fb0861a-7847-4637-8e3e-fab426c4c4db

However, in responsive mode if we open the basic information modal again after Saving Changes, Time Zone field is still showing the old data itself. Please see the below video for details.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/7041610/6b6cfdf3-c295-4d55-bdd4-b82cd5f15e77

**Comment by Cavein254** [comment]
@robertoooc  The pr functions as expected. The timezone changes with the city provided after adding the api key. The css behavior is also satisfactory.

- [x] Enter a few locations that used to crash the page are: vatican, zagreb, podgorica.
- [x] verify that the locations from step 6 don't crash the page and actually change the timezone accordingly.
- [x] Take a look at the css properties of the entire basic information tab, compare it to the original versions and make sure it has the same behavior as before since it was refactored. Make sure to change media sizes to verify this.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13707896/826c9646-0a84-4d54-9cdf-dc7c9b85b1b0

**Review by Renanluizssx — APPROVED** [review]
Hi @robertoooc, i verified again about you told me, and now it´s working correct, very nice.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/653f68d5-3f07-4722-806c-e914ebc21ee4

**Review by peterson337 — APPROVED** [review]
I tested it, and it worked. The layout is responsive, and the CSS is good.

one comunnity

**Review by SaiDeepak-NEU — APPROVED** [review]
I tested the given location timezones in PR and the page did not crash for me. It looked fine and I was able to change my timezone to Zagreb, Vatican and Podgorica successfully without any error.
Screenshot (18)
Screenshot (15)
Screenshot (14)
I also tested the mobile layout and the css looked fine in there as well. Checked responsiveness for iphone, ipad and samsung devices and it looked good.
Screenshot (17)
Screenshot (16)

One thing I noticed though is that, when I click on inspect and the mobile opens up and I click on the basic information tab, a modal opens up with all the fields like first name and last name and email etc. Then when I close the inspect, this modal still stays on the screen. I don't know if this is the expected behaviour, it is defenitely not an error but just letting you know.
Screenshot (19)

**Comment by robertoooc** [comment]
@SRamen1999, 
Do you see any errors in your console ? That media changing size problem is not something I've experienced. 
For the get time zone functionality: 
Check your network requests and look for the get request sent out to the opencage api should look like this https://api.opencagedata.com/geocode/v1/json?key=apikey&q=vatican&pretty=1&limit=1
- what's the status code response for it, it could help in troubleshooting your problem

**Review by yvonnep165 — APPROVED** [review]
Hi @robertoooc , 

I have tested several places for the time zone location including the locations you mentioned that crashed the page previously. I have also tested the css in web, tablet and mobile display. They all look nice and I didn't see bugs. Good jobs!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/103622026/d4482166-ee27-4864-9c14-5946261dd56a

**Review by Sophie-lei — COMMENTED** [review]
Hey! I just tested this PR with the Admin role. I changed the user location to the following three locations and the time zone can be changed automatically. I also viewed the CSS style on mobile and it looks good as well. Please view the video for more information. 
Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/70113822/bc1e3188-e704-4c0b-9716-e318f1a21c1e

**Pull request merged**